File: src/main/java/com/ansh/blog/blogappapis/services/PostService.java
package com.ansh.blog.blogappapis.services;
import java.util.List;
import com.ansh.blog.blogappapis.payloads.PostDto;
import com.ansh.blog.blogappapis.payloads.PostResponse;

public interface PostService {
    //create
    PostDto createPost(PostDto postdto, Integer userId, Integer categoryId);

    //update
    PostDto updatePost(PostDto postdto, Integer postId);

    //delete
    void deletePost(Integer postId);

    //get all posts
    PostResponse getAllPosts(Integer pageNumber, Integer pageSize, String sortBy, String sortDir);

    //get single post
    PostDto getPostById(Integer postId);

    //get all post by category
    List<PostDto> getPostsByCategory (Integer categoryId);

    //get all posts by one user
    PostResponse getPostsByUser (Integer userId,Integer pageNumber,Integer pageSize);

    //search
    List<PostDto> searchPosts(String keyword);
}

File: react-blog-app/src/Services/post_service.js
import { privateAxios } from "./helper_service";
import { myAxios } from "./helper_service";

//Create post function
export const createPost = (postData) => {
  return privateAxios
    .post(
      `/user/${postData.userId}/category/${postData.categoryId}/posts`,
      postData
    )
    .then((resp) => resp.data);
};

//get all posts
export const loadAllPosts = (pageNumber, pageSize) => {
  return myAxios
    .get(
      `/posts?pageNumber=${pageNumber}&pageSize=${pageSize}&sortBy=addedDate&sortDir=desc`
    )
    .then((resp) => resp.data);
};

//load single post of given id
export const loadPost = (postId) => {
  return myAxios.get("/posts/" + postId).then((resp) => resp.data);
};

//upload post banner image
export const uploadPostImage = (image, postId) => {
  let form = new FormData();
  form.append("image", image);
  return privateAxios
    .post(`/post/image/upload/${postId}`, form, {
      headers: {
        "Content-Type": "multipart/form-data",
      },
    })
    .then((resp) => resp.data);
};

//get category wise posts
export const loadPostCategoryWise = (categoryId) => {
  return privateAxios
    .get(`/category/${categoryId}/posts`)
    .then((resp) => resp.data);
};

//get user wise posts
export const loadPostUserWise = (userId) => {
  return privateAxios.get(`/user/${userId}/posts`).then((resp) => resp.data);
};

//delete post by id
export function deletePost(postId) {
  return privateAxios.delete(`/posts/${postId}`).then((resp) => resp.data);
}
